hammer/library/aws/elasticsearch.py

```diff
@@ -0,0 +1,356 @@
+        except Exception:
+            logging.exception(f"Failed to put {self.name} restricted policy")
+            return False
+
+        return True
+
+    def set_logging(self):
+        """
+        
+        :return: 
+        """
+        try:
+            ElasticSearchOperations.set_domain_logging(self.account.client("es"), self.account.client("logs"), self.name)
+        except Exception:
+            logging.exception(f"Failed to enable {self.name} logging")
+            return False
+
+        return True
+
+
+class ESDomainChecker:
+    """
+        Basic class for checking Elasticsearch unencrypted and logging issues in account/region.
+        Encapsulates discovered Elasticsearch domains.
+        """
+
+    def __init__(self, account):
+        """
+        :param account: `Account` instance with Elasticsearch domains to check
+        """
+        self.account = account
+        self.domains = []
+
+    def get_domain(self, id):
+        """
+        :return: `Elasticsearch Domain` by id
+        """
+        for domain in self.domains:
+            if domain.name == id:
+                return domain
+        return None
+
+    def check(self, ids=None):
+        """
+        Walk through Elasticsearch domains in the account/region and put them to `self.domains`.
+
+        :param ids: list with Elasticsearch domain ids to check, if it is not supplied - all Elasticsearch domains must be checked
+
+        :return: boolean. True - if check was successful,
+                          False - otherwise
+        """
+        domain_details = []
+        try:
+            es_client = self.account.client("es")
+            if ids is None:
+                ids = []
+                domain_names_list = es_client.list_domain_names()["DomainNames"]
+                for domain_name in domain_names_list:
+                    ids.append(domain_name["DomainName"])
+
+            if ids is not None:
+                domain_details = es_client.describe_elasticsearch_domains(DomainNames=ids)["DomainStatusList"]
+
+        except ClientError as err:
+            if err.response['Error']['Code'] in ["AccessDenied", "UnauthorizedOperation"]:
+                logging.error(f"Access denied in {self.account} "
+                              f"(ec2:{err.operation_name})")
+            else:
+                logging.exception(f"Failed to describe elasticsearch domains in {self.account}")
+            return False
+
+        for domain_detail in domain_details:
+            is_logging = False
+            domain_encrypted = False
+            domain_name = domain_detail["DomainName"]
+            domain_id = domain_detail["DomainId"]
+            domain_arn = domain_detail["ARN"]
+            encryption_at_rest = domain_detail.get("EncryptionAtRestOptions")
+            node_to_node_encryption = domain_detail.get("NodeToNodeEncryptionOptions")
+            if encryption_at_rest and encryption_at_rest["Enabled"]:
+                domain_encrypted = True
```

Comment:
Should we consider these as different options? If it's encrypted at rest and not in transit, report that it's not encrypted in transit. In the opposite case, report that it's not encrypted at rest.

Reply:
Make changes as per encryption type.